src/Data/Parameterized/Context/Unsafe.hs

```diff
@@ -294,6 +299,24 @@ instance Show (Index ctx tp) where
 
 instance ShowF (Index ctx)
 
+-- | View of indexes as pointing to the last element in the
+-- index range or pointing to an earlier element in a smaller
+-- range.
+data IndexView ctx tp where
+  IndexViewHere :: !(Size ctx) -> IndexView (ctx '::> tp) tp
+  IndexViewThere :: !(Index ctx tp) -> IndexView (ctx '::> u) tp
```

Comment:
This is bike shedding, and the changes look fine.   However, what if we called the constructs `IndexViewLast` and `IndexViewInit` respectively.  They share some similarity with `last` and `init` in `Data.List`.

Reply:
I'm on board